### ISSUE TITLE
Should Referent Subclasses Be Equal To Their Original Class?

### ISSUE THREAD

**Issue by mvoong** [google/j2objc#477]
We've encountered a situation where we're using an object's class as a WeakHashMap key. Further down the line, this object's class is changed when a weak reference is created. This means that we cannot look up the original object by the new class, as it has changed.

Would it be sensible to create an IOSClass's isEquals method to return true in this case?

```java
        Object object = new Object();
        // Object's class in objc is now NSObject
        
        new WeakReference(object);
        // Object's class in objc is now changed to NSObject_ReferentSubclass
        
        assert object.getClass().equals(Object.class); // Fails - should this succeed?
```

**Comment by tomball** [comment]
Yes, your isEquals suggestion seems best.

**Comment by tomball** [comment]
Fixed in 0.9.6.
